### DISCUSSION TITLE
core-daemon not running

### DISCUSSION THREAD

**Discussion by 17lecj** [coreemu/core#656] [Q&A]
Hello, 

I'm installing for the first time CORE in a virtual machine with Ubuntu 20.4.

I executed the following steps:
1.  sudo apt install update
2.  sudo apt upgrade
3.  sudo apt dist-upgrade
4.  git clone https://github.com/coreemu/core.git
5.  cd core
6.  ./setup.sh
7.  inv install

Then tried to run core-daemon executing:
1.  sudo systemctl enable core-daemon
2.  sudo systemctl start core-daemon
3.  core-daemon

The problem is that it shows an error indicating that the port is already used.

Error - core-daemon:cored - error starting main server on: localhost 4038
OSError: [Errno 98] Address already in use.

But if I run the core-gui-legacy, it opens the legacy gui and in terminal it shows that core-daemon was connected.

I was hoping someone could help me figuring out what's wrong with current the gui or if I'm installing something incorrectly.

Thank you!

**Comment by bharnden** [comment]
If you are already running the service, then that starts `core-daemon`, so you would not need to do `3` and run `core-daemon` again. So that makes sense that it would complain about the port already being used.

Since you are running the service and the `core-daemon` is already running, then the GUI can connect. So it all sounds about right.

**Reply by 17lecj** [reply]
Got it! It makes sense. 

I'm starting to learn how to use this tool. How would I know that the gui is working? 
I'm linking two routers. If I hit run, the legacy version shows a red rectangle turning green for each item and enables the two node tool. Using the current gui, the only thing that changes is the Run button to Stop. The two node tool doesn't appear.

Thanks for your help!

**Comment by bharnden** [comment]
The new GUI would not turn to a stop icon and present an error about the failure to start. So turning to stop without an error, is a good outcome.

**Comment by mdzahidul-islam** [comment]
Hey, I am also new and have followed the same installation guide. Everything is fine. 
However, after running the season in GUI, when I try to ping a address, the _run command_ hangs. Also, the run window looks different than the GUI I found on some youtube tutorials.
Is there any suggestions? 
Thanks in advance.

**Comment by Pratyay211** [comment]
Hello,

I was currently trying to install CORE on my virtual ubuntu version 20.04.5.

I have executed the following command:

git clone https://github.com/coreemu/core.git
cd core
./setup.sh
inv install

I am getting the following error:

installing core virtual environment ... |Encountered a bad command exit code! 
 
Command: 'cd daemon && poetry install --no-dev'
 
Exit code: 1
 
Stdout:
 
      1179│                 output = subprocess.check_output(
      1180│                     cmd, stderr=subprocess.STDOUT, **kwargs
      1181│                 )
      1182│         except CalledProcessError as e:
    → 1183│             raise EnvCommandError(e, input=input_)
      1184│ 
      1185│         return decode(output)
      1186│ 
      1187│     def execute(self, bin, *args, **kwargs):
 
 
Stderr:
 


Can anyone suggest to me what I should do?

**Reply by bharnden** [reply]
This will happen usually related to trying to build a python dependency from source, and you are missing system packages to build it.

You can run the command given manually to see the specific error, or run the install again with the -v to see all output

**Reply by Pratyay211** [reply]
@bharnden Thank you for your quick response. After running the install again with -v the output us like:

pratyay@ubuntu:~/core$ inv install -v
installing core locally: False
installing core with prefix: /usr/local
[sudo] password for pratyay: 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named core
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'core'

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
automake is already the newest version (1:1.16.1-4ubuntu6).
ethtool is already the newest version (1:5.4-1).
gcc is already the newest version (4:9.3.0-1ubuntu2).
iproute2 is already the newest version (5.5.0-1ubuntu1).
pkg-config is already the newest version (0.29.1-0ubuntu4).
libev-dev is already the newest version (1:4.31-1).
nftables is already the newest version (0.9.3-2).
tk is already the newest version (8.6.9+1).
bash is already the newest version (5.0-6ubuntu1.2).
python3-tk is already the newest version (3.8.10-0ubuntu1~20.04).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Requirement already satisfied: grpcio==1.27.2 in /home/pratyay/.local/lib/python3.8/site-packages (1.27.2)
Requirement already satisfied: grpcio-tools==1.27.2 in /home/pratyay/.local/lib/python3.8/site-packages (1.27.2)
Requirement already satisfied: six>=1.5.2 in /home/pratyay/.local/lib/python3.8/site-packages (from grpcio==1.27.2) (1.16.0)
Requirement already satisfied: protobuf>=3.5.0.post1 in /usr/lib/python3/dist-packages (from grpcio-tools==1.27.2) (3.6.1)
Bootstrapping the autoconf system...
(Messages below about copying and installing files are normal.)
(1/4) Running aclocal...
(2/4) Running autoheader...
(3/4) Running automake...
(4/4) Running autoconf...

You are now ready to run "./configure".
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '1000' is supported by ustar format... yes
checking whether GID '1000' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking for gawk... (cached) mawk
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking whether make sets $(MAKE)... (cached) yes
checking for ranlib... ranlib
checking for a sed that does not truncate output... /usr/bin/sed
checking for NgMkSockNode in -lnetgraph... no
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/mount.h usability... yes
checking sys/mount.h presence... yes
checking for sys/mount.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking termios.h usability... yes
checking termios.h presence... yes
checking for termios.h... yes
checking for unistd.h... (cached) yes
checking for inline... inline
checking for int32_t... yes
checking for pid_t... yes
checking for size_t... yes
checking for ssize_t... yes
checking for uint32_t... yes
checking for uint8_t... yes
checking vfork.h usability... no
checking vfork.h presence... no
checking for vfork.h... no
checking for fork... no
checking for vfork... yes
checking for working vfork... (cached) yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible realloc... yes
checking for atexit... yes
checking for dup2... yes
checking for gettimeofday... yes
checking for memset... no
checking for socket... yes
checking for strerror... yes
checking for uname... yes
checking for a Python interpreter with version >= 3.6... python3
checking for python3... /usr/bin/python3
checking for python3 version... 3.8
checking for python3 platform... linux
checking for python3 script directory... ${prefix}/lib/python3.8/site-packages
checking for python3 extension module directory... ${exec_prefix}/lib/python3.8/site-packages
checking for sysctl... /usr/sbin
checking for nft... /usr/sbin
checking for ip... /usr/sbin
checking for tc... /usr/sbin
checking for ethtool... /usr/sbin
checking for mount... /usr/bin
checking for umount... /usr/bin
checking for convert... yes
checking for ovs-vsctl... no
configure: WARNING: Could not locate ovs-vsctl cannot use OVS mode
checking for ovs-ofctl... no
configure: WARNING: Could not locate ovs-ofctl cannot use OVS mode
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libev... no
did not find libev using pkconfig...
checking for ev_set_allocator in -lev... yes
found libev OK
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating gui/version.tcl
config.status: creating gui/Makefile
config.status: creating gui/icons/Makefile
config.status: creating man/Makefile
config.status: creating docs/Makefile
config.status: creating daemon/Makefile
config.status: creating daemon/doc/Makefile
config.status: creating daemon/doc/conf.py
config.status: creating daemon/proto/Makefile
config.status: creating netns/Makefile
config.status: creating netns/version.h
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
------------------------------------------------------------------------
core 8.2.0 Configuration:

Build:
  Host System Type:           
  C Compiler and flags:       gcc  -O3 -Werror -Wall -D_GNU_SOURCE
  Prefix:                     /usr/local
  Exec Prefix:                ${prefix}

GUI:
  GUI path:                   ${prefix}/lib/core
  GUI config:                 $${HOME}/.core

Daemon:
  Daemon path:                ${exec_prefix}/bin
  Daemon config:              /etc/core
  Python:                     /usr/bin/python3

Features to build:
  Build GUI:                  yes
  Build Daemon:               yes
  Documentation:              no

------------------------------------------------------------------------
On this platform you should run core-gui as a normal user.
------------------------------------------------------------------------
creating file gui/core-gui-legacy from gui/core-gui-legacy.in
creating file daemon/core/constants.py from daemon/core/constants.py.in
creating file netns/setup.py from netns/setup.py.in
make  all-recursive
make[1]: Entering directory '/home/pratyay/core'
Making all in gui
make[2]: Entering directory '/home/pratyay/core/gui'
Making all in icons
make[3]: Entering directory '/home/pratyay/core/gui/icons'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/pratyay/core/gui/icons'
make[3]: Entering directory '/home/pratyay/core/gui'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/pratyay/core/gui'
make[2]: Leaving directory '/home/pratyay/core/gui'
Making all in daemon
make[2]: Entering directory '/home/pratyay/core/daemon'
Making all in proto
make[3]: Entering directory '/home/pratyay/core/daemon/proto'
/usr/bin/python3 -m grpc_tools.protoc -I . --python_out=.. core/api/grpc/*.proto
/usr/bin/python3 -m grpc_tools.protoc -I . --grpc_python_out=.. core/api/grpc/core.proto
make[3]: Leaving directory '/home/pratyay/core/daemon/proto'
make[3]: Entering directory '/home/pratyay/core/daemon'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/pratyay/core/daemon'
make[2]: Leaving directory '/home/pratyay/core/daemon'
Making all in netns
make[2]: Entering directory '/home/pratyay/core/netns'
gcc -DHAVE_CONFIG_H -I. -I..    -Wall -fno-strict-aliasing -O3 -g  -O3 -Werror -Wall -D_GNU_SOURCE -MT vnoded_main.o -MD -MP -MF .deps/vnoded_main.Tpo -c -o vnoded_main.o vnoded_main.c
gcc -DHAVE_CONFIG_H -I. -I..    -Wall -fno-strict-aliasing -O3 -g  -O3 -Werror -Wall -D_GNU_SOURCE -MT vcmd_main.o -MD -MP -MF .deps/vcmd_main.Tpo -c -o vcmd_main.o vcmd_main.c
mv -f .deps/vnoded_main.Tpo .deps/vnoded_main.Po
gcc -DHAVE_CONFIG_H -I. -I..    -Wall -fno-strict-aliasing -O3 -g  -O3 -Werror -Wall -D_GNU_SOURCE -MT netns_main.o -MD -MP -MF .deps/netns_main.Tpo -c -o netns_main.o netns_main.c
mv -f .deps/vcmd_main.Tpo .deps/vcmd_main.Po
gcc -Wall -fno-strict-aliasing -O3 -g  -O3 -Werror -Wall -D_GNU_SOURCE   -o vnoded vnode_msg.o vnode_cmd.o vnode_chnl.o vnode_io.o vnoded_main.o vnode_server.o netns.o -lev 
mv -f .deps/netns_main.Tpo .deps/netns_main.Po
gcc -Wall -fno-strict-aliasing -O3 -g  -O3 -Werror -Wall -D_GNU_SOURCE   -o vcmd vnode_msg.o vnode_cmd.o vnode_chnl.o vnode_io.o vcmd_main.o vnode_client.o -lev 
gcc -Wall -fno-strict-aliasing -O3 -g  -O3 -Werror -Wall -D_GNU_SOURCE   -o netns netns_main.o netns.o  
make[2]: Leaving directory '/home/pratyay/core/netns'
make[2]: Entering directory '/home/pratyay/core'
make[2]: Leaving directory '/home/pratyay/core'
make[1]: Leaving directory '/home/pratyay/core'
Making install in gui
make[1]: Entering directory '/home/pratyay/core/gui'
Making install in icons
make[2]: Entering directory '/home/pratyay/core/gui/icons'
make[3]: Entering directory '/home/pratyay/core/gui/icons'
make[3]: Nothing to be done for 'install-exec-am'.
 /usr/bin/mkdir -p '/usr/local/share/core/icons/normal'
 /usr/bin/install -c -m 644 normal/text.gif normal/wlan.gif normal/hub.gif normal/router_red.gif normal/router.gif normal/router_green.gif normal/rj45.gif normal/antenna.gif normal/core-icon.png normal/document-properties.gif normal/oval.gif normal/host.gif normal/thumb-unknown.gif normal/router_black.gif normal/core-logo-275x75.gif normal/simple.xbm normal/tunnel.gif normal/pc.gif normal/gps-diagram.xbm normal/lanswitch.gif normal/OVS.gif normal/core-icon.xbm normal/router_purple.gif normal/router_yellow.gif normal/ap.gif normal/mdr.gif '/usr/local/share/core/icons/normal'
 /usr/bin/mkdir -p '/usr/local/share/core/icons/svg'
 /usr/bin/install -c -m 644 svg/router_yellow.svg svg/ap.svg svg/mdr.svg svg/rj45.svg svg/vlan.svg svg/hub.svg svg/router_red.svg svg/router.svg svg/router_green.svg svg/start.svg svg/tunnel.svg svg/router_black.svg svg/otr.svg svg/lanswitch.svg svg/OVS.svg svg/cel.svg svg/router_purple.svg '/usr/local/share/core/icons/svg'
 /usr/bin/mkdir -p '/usr/local/share/core/icons/tiny'
 /usr/bin/install -c -m 644 tiny/text.gif tiny/blank.gif tiny/button.play.gif tiny/ping.gif tiny/folder.gif tiny/wlan.gif tiny/router_red.gif tiny/arrow.down.gif tiny/arrow.gif tiny/script_play.gif tiny/router_green.gif tiny/link.gif tiny/start.gif tiny/plot.gif tiny/stock_connect.gif tiny/select.gif tiny/document-properties.gif tiny/oval.gif tiny/run.gif tiny/marker.gif tiny/host.gif tiny/document-save.gif tiny/delete.gif tiny/script_stop.gif tiny/router_black.gif tiny/stop.gif tiny/hub.gif tiny/trace.gif tiny/edit-delete.gif tiny/document-new.gif tiny/OVS.gif tiny/stock_disconnect.gif tiny/mdr.gif tiny/tunnel.gif tiny/arrow.up.gif tiny/pc.gif tiny/router.gif tiny/moboff.gif tiny/lanswitch.gif tiny/mobility.gif '/usr/local/share/core/icons/tiny'
 /usr/bin/install -c -m 644 tiny/script_pause.gif tiny/twonode.gif tiny/fileopen.gif tiny/observe.gif tiny/router_purple.gif tiny/rectangle.gif tiny/router_yellow.gif tiny/ap.gif tiny/cel.gif tiny/button.stop.gif tiny/eraser.gif tiny/view-refresh.gif tiny/rj45.gif '/usr/local/share/core/icons/tiny'
 /usr/bin/mkdir -p '/usr/local/share/applications'
 /usr/bin/install -c -m 644 core-gui.desktop '/usr/local/share/applications'
 /usr/bin/mkdir -p '/usr/local/share/pixmaps'
 /usr/bin/install -c -m 644 core-gui.xpm '/usr/local/share/pixmaps'
make[3]: Leaving directory '/home/pratyay/core/gui/icons'
make[2]: Leaving directory '/home/pratyay/core/gui/icons'
make[2]: Entering directory '/home/pratyay/core/gui'
make[3]: Entering directory '/home/pratyay/core/gui'
 /usr/bin/mkdir -p '/usr/local/bin'
 /usr/bin/install -c core-gui-legacy '/usr/local/bin'
 /usr/bin/mkdir -p '/usr/local/lib/core'
 /usr/bin/install -c -m 644 traffic.tcl wlan.tcl core.tcl cfgparse.tcl annotations.tcl plugins.tcl gpgui.tcl help.tcl ipv6.tcl version.tcl tooltips.tcl api.tcl nodecfg.tcl wlanscript.tcl debug.tcl filemgmt.tcl topogen.tcl mobility.tcl ipv4.tcl services.tcl initgui.tcl widget.tcl canvas.tcl util.tcl exceptions.tcl ns2imunes.tcl exec.tcl graph_partitioning.tcl editor.tcl nodes.tcl linkcfg.tcl '/usr/local/lib/core'
 /usr/bin/mkdir -p '/usr/local/lib/core/addons'
 /usr/bin/install -c -m 644 addons/ipsecservice.tcl '/usr/local/lib/core/addons'
 /usr/bin/mkdir -p '/usr/local/share/core/examples/configs'
 /usr/bin/install -c -m 644 configs/sample7-emane-ieee80211abg.imn configs/sample4.scen configs/sample9-vpn.imn configs/sample1.scen configs/sample4-nrlsmf.imn configs/sample2-ssh.imn configs/sample10-kitchen-sink.imn configs/sample5-mgen.imn configs/sample6-emane-rfpipe.imn configs/sample1-bg.gif configs/sample8-ipsec-service.imn configs/sample4-bg.jpg configs/sample1.imn configs/sample11-sdn.imn configs/sample3-bgp.imn '/usr/local/share/core/examples/configs'
make[3]: Leaving directory '/home/pratyay/core/gui'
make[2]: Leaving directory '/home/pratyay/core/gui'
make[1]: Leaving directory '/home/pratyay/core/gui'
Making install in daemon
make[1]: Entering directory '/home/pratyay/core/daemon'
Making install in proto
make[2]: Entering directory '/home/pratyay/core/daemon/proto'
make[3]: Entering directory '/home/pratyay/core/daemon/proto'
make[3]: Nothing to be done for 'install-exec-am'.
make[3]: Nothing to be done for 'install-data-am'.
make[3]: Leaving directory '/home/pratyay/core/daemon/proto'
make[2]: Leaving directory '/home/pratyay/core/daemon/proto'
make[2]: Entering directory '/home/pratyay/core/daemon'
make[3]: Entering directory '/home/pratyay/core/daemon'
make[3]: Nothing to be done for 'install-exec-am'.
make[3]: Nothing to be done for 'install-data-am'.
make[3]: Leaving directory '/home/pratyay/core/daemon'
make[2]: Leaving directory '/home/pratyay/core/daemon'
make[1]: Leaving directory '/home/pratyay/core/daemon'
Making install in netns
make[1]: Entering directory '/home/pratyay/core/netns'
make[2]: Entering directory '/home/pratyay/core/netns'
 /usr/bin/mkdir -p '/usr/local/bin'
  /usr/bin/install -c vnoded vcmd netns '/usr/local/bin'
make[2]: Nothing to be done for 'install-data-am'.
make[2]: Leaving directory '/home/pratyay/core/netns'
make[1]: Leaving directory '/home/pratyay/core/netns'
make[1]: Entering directory '/home/pratyay/core'
make[2]: Entering directory '/home/pratyay/core'
make[2]: Nothing to be done for 'install-exec-am'.
make[2]: Nothing to be done for 'install-data-am'.
make[2]: Leaving directory '/home/pratyay/core'
make[1]: Leaving directory '/home/pratyay/core'
Installing dependencies from lock file

Package operations: 1 install, 0 updates, 0 removals

  • Installing pyproj (2.6.1.post1)

  EnvCommandError

  Command ['/home/pratyay/.cache/pypoetry/virtualenvs/core-Lk8DyInX-py3.8/bin/pip', 'install', '--no-deps', 'file:///home/pratyay/.cache/pypoetry/artifacts/41/e5/59/9ef614ae56728577632d1b692d42e226ec86fd698af527673803aa1d31/pyproj-2.6.1.post1.tar.gz'] errored with the following return code 1, and output: 
  Processing /home/pratyay/.cache/pypoetry/artifacts/41/e5/59/9ef614ae56728577632d1b692d42e226ec86fd698af527673803aa1d31/pyproj-2.6.1.post1.tar.gz
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'error'
    error: subprocess-exited-with-error
    
    × Getting requirements to build wheel did not run successfully.
    │ exit code: 1
    ╰─> [1 lines of output]
        proj executable not found. Please set the PROJ_DIR variable.For more information see: https://pyproj4.github.io/pyproj/stable/installation.html
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  

  at ~/.local/lib/python3.8/site-packages/poetry/utils/env.py:1183 in _run
      1179│                 output = subprocess.check_output(
      1180│                     cmd, stderr=subprocess.STDOUT, **kwargs
      1181│                 )
      1182│         except CalledProcessError as e:
    → 1183│             raise EnvCommandError(e, input=input_)
      1184│ 
      1185│         return decode(output)
      1186│ 
      1187│     def execute(self, bin, *args, **kwargs):

pratyay@ubuntu:~/core$
